src/memoize-one.ts

```diff
@@ -1,40 +1,56 @@
 import areInputsEqual from './are-inputs-equal';
 
 // Using ReadonlyArray<T> rather than readonly T as it works with TS v3
-export type EqualityFn = (newArgs: any[], lastArgs: any[]) => boolean;
-
-function memoizeOne<
-  // Need to use 'any' rather than 'unknown' here as it has
-  // The correct Generic narrowing behaviour.
-  ResultFn extends (this: any, ...newArgs: any[]) => ReturnType<ResultFn>
->(resultFn: ResultFn, isEqual: EqualityFn = areInputsEqual): ResultFn {
-  let lastThis: unknown;
-  let lastArgs: unknown[] = [];
-  let lastResult: ReturnType<ResultFn>;
-  let calledOnce: boolean = false;
+export type EqualityFn<TFunc extends (...args: any[]) => any> = (
```

Comment:
The types of the EqualityFn are now based on the function that it is being used for